src/main/java/com/google/devtools/build/lib/bazel/repository/PatchUtil.java

```diff
@@ -229,8 +237,35 @@ private static void writeFile(Path file, List<String> content) throws IOExceptio
     FileSystemUtils.writeLinesAs(file, StandardCharsets.UTF_8, content);
   }
 
+  private static boolean getReadPermission(int permission) {
+    // Parse read permission from posix file permission notation
+    return permission / 4 % 2 == 1;
```

Comment:
We can use bit-operators here to test for the bits in the permission bitmask :)

Executable is `0b001 = 1`
Writable is `0b010 = 2`
Readable is `0b100 = 4`

So we can replace this with:

```java
return permission & 4;
```

Reply:
Thanks, I'll use 
```java 
return (permission & 4) == 4
```